### PR TITLE
[MINOR]: Limit stream replace with slice

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

File: datafusion/physical-plan/src/limit.rs [hunk 1]
```diff
@@ -512,21 +511,8 @@ impl LimitStream {
...
-            Some(
-                RecordBatch::try_new_with_options(
-                    batch.schema(),
-                    limited_columns,
-                    &options,
-                )
-                .unwrap(),
-            )
+            // It is guaranteed that batch_rows is <= batch.num_rows
+            Some(batch.slice(0, batch_rows))
```
Comment (comphead) [inline datafusion/physical-plan/src/limit.rs:515]: may be we can put a condition with `unreacheable!` ?

Comment (mustafasrepo) [reply datafusion/physical-plan/src/limit.rs:515]: did so

**Review by comphead — APPROVED** [review]
lgtm thanks @mustafasrepo

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Thank you @mustafasrepo

**Pull request merged**